presto-hive/src/main/java/io/prestosql/plugin/hive/HiveErrorCode.java

```diff
@@ -64,6 +64,8 @@
     HIVE_CORRUPTED_COLUMN_STATISTICS(37, EXTERNAL),
     HIVE_EXCEEDED_SPLIT_BUFFERING_LIMIT(38, USER_ERROR),
     HIVE_UNKNOWN_COLUMN_STATISTIC_TYPE(39, INTERNAL_ERROR),
+    HIVE_TABLE_LOCK_NOT_ACQUIRED(40, EXTERNAL),
+    HIVE_TABLE_TXN_ABORTED_BY_MS(41, EXTERNAL)
```

Comment:
```suggestion
    HIVE_TABLE_TRANSACTION_ABORTED_BY_METASTORE(41, EXTERNAL)
```

Reply:
Also, this can be considered EXTERNAL or USER_ERROR -- if user provided too low heartbeat interval value.

(let's keep it EXTERNAL but let's get appropriate hearbeat interval from transaction manager at runtime)